Mage/src/main/java/mage/game/permanent/PermanentImpl.java

```diff
@@ -1351,7 +1351,23 @@ public boolean cantBeAttachedBy(MageObject attachment, Ability source, Game game
                 return !ability.getDoesntRemoveControlled() || isControlledBy(game.getControllerId(attachment.getId()));
             }
         }
-        return game.getContinuousEffects().preventedByRuleModification(new StayAttachedEvent(this.getId(), attachment.getId(), source), null, game, silentMode);
+
+        boolean canAttach = true;
+        Permanent attachmentPermanent = game.getPermanent(attachment.getId());
```

Comment:
Why you load attachment to attachmentPermanent? It’s same object already.

Reply:
Code brevity; it's more readable in the code to use a variable called `attachmentPermanent` and use that than it is to write `game.getPermanent(attachment.getId())` each time

Reply:
Nope, I’m talking about “MageObject attachment” from method’s params. It’s same object, so why not to use it? Perm = ((Permanent) attachment).

Reply:
Also be careful — that’s method used for aura cards too, not for permanents only. So it must be compatible for both use cases.

Card usage examples:
![IMG_0202](https://github.com/magefree/mage/assets/8344157/e9dec04e-1d12-4ffc-8507-7c6f79bae67e)
![IMG_0204](https://github.com/magefree/mage/assets/8344157/e4b240af-f618-47c1-a4d6-26f5eb1fb323)

Reply:
I think this is OK - I tested with casting an aura, making the aura's target illegal, and then resolving the aura. The outcome was, as expected, the aura going to the graveyard. Will add a test for futureproofing

Reply:
Current code with game.getPermanent checks validation for permanents only (e.g. aura on battlefield try to reattach). And it will ignore card (e.g. aura moves from hand to battlefield without cast/stack).

Reply:
"Sneaking" an aura onto the battlefield with [Eureka](https://scryfall.com/card/leg/185/eureka) gives no opportunity for a permanent to become an illegal target before the user selects the target for the aura to attach to. In the case that there are no legal targets, the aura goes to the graveyard as it is unattached. Is that what you mean or did you have another interaction in mind?

Reply:
See code screens above. Example with Unfinished Business:

> Return target creature card from your graveyard to the battlefield, then return up to two target Aura and/or Equipment cards from your graveyard to the battlefield attached to that creature. (If the Auras can’t enchant that creature, they remain in your graveyard.)

Reply:
Ah yes, of course. I have tested and can confirm that the interaction works as expected (the aura does not leave the graveyard). Will add a test for that case to this PR